pageserver/src/tenant/mgr.rs

```diff
@@ -275,7 +349,7 @@ pub(crate) fn schedule_local_tenant_processing(
     conf: &'static PageServerConf,
     tenant_id: TenantId,
     tenant_path: &Path,
-    generation: Generation,
+    location_conf: LocationConf,
```

Comment:
Would like enforcement in the type system that we're in Attached mode at this point.

Reply:
This one is kind of a tradeoff.  I would also kind of like for the attached states to only see the `AttachedLocationConfig`, but to do it that way, one has to do one of the following:
- move the TenantConfOpt into the LocationMode part, such that each LocationMode has the tenant config
- Make Tenant carry its location config and its TenantConfOpt separately, requiring two args here, two Mutexes in the struct, etc.

Reply:
I think I like 3 best.

Reply:
In 117213a2d1ed198b5388b5090a85de28c10d7a90, Tenants now only get an AttachedTenantConf, which is just the subset of LocationConf that they need